linux_os/guide/services/sssd/sssd_enable_pam_services/ansible/shared.yml

```diff
@@ -0,0 +1,41 @@
+# platform = multi_platform_all
+# reboot = false
+# strategy = configure
+# complexity = low
+# disruption = medium
+
+- name: {{{ rule_title }}} - Find all the conf files inside the /etc/sssd/conf.d/ directory
+  ansible.builtin.find:
+    paths:
+    - "/etc/sssd/conf.d/"
+    patterns: "*.conf"
+  register: sssd_conf_d_files
+
+- name: {{{ rule_title }}} - Modify lines in files in the /etc/sssd/conf.d/ directory
+  ansible.builtin.replace:
+    path: "{{ item }}"
+    regexp: '^(\s*\[sssd\].*(?:\n\s*[^[\s].*)*\n\s*services\s*=(?!.*\bpam\b).*)$'
+    replace: '\1,pam'
+  with_items: "{{ sssd_conf_d_files.files | map(attribute='path') }}"
+  register: modify_lines_sssd_conf_d_files
+
+- name: {{{ rule_title }}} - Find /etc/sssd/sssd.conf
+  ansible.builtin.stat:
+    path: /etc/sssd/sssd.conf
+  register: sssd_conf_file
+
+- name: {{{ rule_title }}} - Modify lines in /etc/sssd/sssd.conf
+  ansible.builtin.replace:
+    path: "/etc/sssd/sssd.conf"
+    regexp: '^(\s*\[sssd\].*(?:\n\s*[^[\s].*)*\n\s*services\s*=(?!.*\bpam\b).*)$'
+    replace: '\1,pam'
+  register: modify_lines_sssd_conf_file
+  when: sssd_conf_file.stat.exists
+
+- name: {{{ rule_title }}} - Insert entry to /etc/sssd/sssd.conf
+  ini_file:
+    path: /etc/sssd/sssd.conf
+    section: sssd
+    option: services
+    value: pam
```

Comment:
There is an issue with this last task. It is here to ensure a case where the services line is not present in [sssd]. However in most cases, if not all cases, the services is already there and commonly would include more services instead of only `pam`.
For example, if the line is:
`services = nss,pam`
This last task will remove `nss` and leave the line as `services = pam`. This is not desired.

I believe the way to solve this is creating another task in check mode only to test if there is already a line with `services = *` and use the result in this last task.

Reply:
I believe the same logic used here can be useful for this case:
https://github.com/ComplianceAsCode/content/blob/master/shared/macros/10-ansible.jinja#L1435-L1462